modules/src/wake.py

```diff
@@ -0,0 +1,59 @@
+import config
+import modules
+from templates.text import TextTemplate
+
+from datetime import datetime, timedelta
+from time import strptime
+
+def calculate_time(input_time=None): #Input must be in hh:mm format
+    hours = [9, 7, 6, 4] 
+    minutes = [0, 30, 0, 30] # The formula is based on www.sleepyti.me
+    answer = []
+    user_input = []
+    _hour = 0
+    _minute = 0
+
+    if not input_time: #if user did not input any time 
+        user_input.append(datetime.now().hour)
+        user_input.append(datetime.now().minute)
+    else:
+        constructed_time = strptime(input_time, '%H:%M') # extract time from string to tuple of datetime format
+        user_input.append(constructed_time[3])
+        user_input.append(constructed_time[4])
+
+    for i in range(4):
```

Comment:
range -> xrange
i -> index (or something similar, idea is to make the variable name verbose)